lib/index.js

```diff
@@ -36,13 +38,22 @@ var method = module.exports = (obj, key, method) => {
   }
   flag.add(key);
 
-  Object.defineProperty(obj, key, {
-    writable: true,
-    configurable: true,
-    enumerable: true,
-    value: method
-  });
-
+  if (!method && extend) {
+    // method->null use extend
+    Object.defineProperty(obj, key, {
+      configurable: true,
+      enumerable: true,
+      get: extend.getter || noop,
+      set: extend.setter || noop,
+    });
+  } else {
+    Object.defineProperty(obj, key, {
+      writable: true,
+      configurable: true,
+      enumerable: true,
+      value: method,
+    });
+  }
```

Comment:
Logic here could be cleaned up a bit. Assign the descriptor object at first, and then give it `value`, `get`, and `set` properties depending on what's defined in `method` and `extend`